app/models/miq_report/search.rb

```diff
@@ -108,17 +110,24 @@ def paged_view_search(options = {})
       options[:no_sort] = true
       self.extras[:target_ids_for_paging] = attrs.delete(:target_ids_for_paging)
     end
-    build_table(search_results, db, options)
+    build_table(filtered_results, db, options)
 
     # build a hash of target objects for UI since we already have them
     if options[:targets_hash]
       attrs[:targets_hash] = {}
-      search_results.each { |obj| attrs[:targets_hash][obj.id] = obj }
+      filtered_results.each { |obj| attrs[:targets_hash][obj.id] = obj }
     end
     attrs[:apply_sortby_in_search] = !!order
     self.extras[:attrs_for_paging] = attrs.merge(:targets_hash => nil) unless self.extras[:target_ids_for_paging].nil?
 
     _log.debug("Attrs: #{attrs.merge(:targets_hash => "...").inspect}")
     return table, attrs
   end
+
+  private
+
+  def filter_results(results, supports_filter)
+    return results if supports_filter.nil?
+    results.select { |result| result.send(supports_filter) }
```

Comment:
and how is `supports_filter` build? Is `options` in `application_controler` filled from user input? If thats the case, this would be a security concern.

So, I'd prefer using `result.supports?(feature.to_sym)`